c10/cuda/test/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@ if(BUILD_TEST)
     get_filename_component(test_file_name ${test_src} NAME_WE)
     set(test_name "c10_cuda_${test_file_name}")
     add_executable(${test_name} "${test_src}")
-    target_link_libraries(${test_name} c10_cuda gtest_main)
+    target_link_libraries(${test_name} ${C10_CUDA_LIB} ${C10_LIB} gtest_main)
```

Comment:
This change will affect both new wheels and old ? before we used to link only with
``
c10_cuda
`` 
now with both 
``
c10_cuda c10
``

Reply:
c10_cuda contains c10, so this is effectively a no-op.

The error that led to this is that we were missing symbols in c10, so we explicitly link it here. My guess is that when we link from the external wheel, the associative libraries are not included (in this case you can't get to c10 from c10_cuda).

Reply:
Here can we add an if statement like this to avoid changing the default current build ?
```
if(BUILD_LIBTORCHLESS)
target_link_libraries(${test_name} ${C10_CUDA_LIB} ${C10_LIB} gtest_main)
else()
target_link_libraries(${test_name} ${C10_CUDA_LIB} gtest_main)
endif()